use existing app.server

laggingreflex opened the pull request:
Checks to see if `app.server` exists, if so, uses that, if not, creates one.

Wraps `app.listen` with warning *only* if `app.server` was created here.

Doesn't throw error if `app.server` exists (this is causing to fail 1 test in `attach.test.js` 'should not alter a koa app that already has .server')

mattstyles commented:
Yeah, looks good to me. Can you fix up the test or remove it and I'll merge it in!

Thanks

laggingreflex commented:
Updated. It also checks now whether `app.server` if already exists is actually an http server. (`app.server.constructor.name` should be "Server"), just in case.

Updated tests.

mattstyles commented:
Thats awesome!

Thanks!

laggingreflex commented:
publish to npm please

mattstyles commented:
Yep, all in 4.1.0. I was going to add some other stuff but that'll take longer so it's up now.
